src/Uno.Extensions.Navigation.UI/RouteResolverDefault.cs

```diff
@@ -45,13 +45,54 @@ IViewRegistry views
 			return null;
 		}
 
-		if (Logger.IsEnabled(LogLevel.Warning)) Logger.LogWarningMessage($"For better performance (avoid reflection), create mapping for for path '{path}', view '{view?.Name}', view model '{viewModel?.Name}'");
+		// Trim any qualifiers
+		path = path.ExtractBase(out _, out _);
 
+		// If no path is provided, attempt to get a path
+		// from the view or viewmodel type provided
 		if (string.IsNullOrWhiteSpace(path))
 		{
 			path = PathFromTypes(view, viewModel);
 		}
 
+		// If path is still null, we can't build a mapping, so just return
+		if (path is null ||
+			string.IsNullOrWhiteSpace(path))
+		{
+			return null;
+		}
+
+		// Attempt to find a viewmap to build the routeinfo from
+		var viewMap = ViewMaps.Items.FirstOrDefault(x =>
+		{
+			var mapView = x.View ?? x.ViewSelector?.Invoke();
+			var mapViewModel = x.ViewModel;
+			var mapPath = PathFromTypes(mapView, mapViewModel);
+			return mapPath == path;
+		});
+		if (viewMap is not null)
+		{
+			var viewFunc = (viewMap.View is not null) ?
+										() => viewMap.View :
+										viewMap.ViewSelector;
+			var defaultMapFromViewMap = new RouteInfo(
+												Path: path,
+												View: viewFunc,
+												ViewAttributes: viewMap.ViewAttributes,
+												ViewModel: viewMap.ViewModel,
+												Data: viewMap.Data?.Data,
+												ToQuery: viewMap?.Data?.UntypedToQuery,
+												FromQuery: viewMap?.Data?.UntypedFromQuery,
+												ResultData: viewMap?.ResultData);
+
+
+			//new RouteInfo(path, View: () => view, ViewModel: viewModel);
```

Comment:
Why this is commented here? not needed anymore or needs more details as a comment in the code for a specific current blocker?

Reply:
Removed